packages/scenes/src/components/layout/CSSGrid/SceneCSSGridLayout.tsx

```diff
@@ -0,0 +1,120 @@
+import { css, CSSObject } from '@emotion/css';
+import { config } from '@grafana/runtime';
+import React, { ComponentType, CSSProperties, useMemo } from 'react';
+
+import { SceneObjectBase } from '../../../core/SceneObjectBase';
+import { SceneComponentProps, SceneLayout, SceneObjectState, SceneObject } from '../../../core/types';
+
+export interface SceneFlexItemStateLike extends SceneObjectState {}
+
+export interface SceneFlexItemLike extends SceneObject<SceneFlexItemStateLike> {}
+
+interface SceneCSSGridLayoutState extends SceneObjectState, SceneCSSGridItemPlacement {
+  children: Array<SceneFlexItemLike | SceneObject>;
+}
+
+export class SceneCSSGridLayout extends SceneObjectBase<SceneCSSGridLayoutState> implements SceneLayout {
+  public static Component = SceneCSSGridLayoutRenderer;
+
+  public isDraggable(): boolean {
+    return false;
+  }
+}
+
+function SceneCSSGridLayoutRenderer({ model, parentState }: SceneCSSGridItemRenderProps<SceneCSSGridLayout>) {
+  const { children, isHidden } = model.useState();
+  const style = useLayoutStyle(model.state);
+
+  if (isHidden) {
+    return null;
+  }
+
+  return (
+    <div className={style}>
+      {children.map((item) => {
+        const Component = item.Component as ComponentType<SceneCSSGridItemRenderProps<SceneObject>>;
+        return <Component key={item.state.key} model={item} parentState={model.state} />;
+      })}
+    </div>
+  );
+}
+
+export interface SceneCSSGridItemPlacement {
+  rows: CSSProperties['gridTemplateRows'];
+  columns: CSSProperties['gridTemplateColumns'];
+  rowGap?: CSSProperties['rowGap'];
+  columnGap?: CSSProperties['columnGap'];
+  justifyItems?: CSSProperties['justifyItems'];
+  alignItems?: CSSProperties['alignItems'];
+  justifyContent?: CSSProperties['justifyContent'];
```

Comment:
```suggestion
export interface SceneCSSGridItemPlacement {
  templateRows: CSSProperties['gridTemplateRows'];
  templateColumns: CSSProperties['gridTemplateColumns'];
  rowGap?: number; // This should probably be in Grafana theme grid units 
  columnGap?: number; This should probably be in Grafana theme grid units 
  justifyItems?: CSSProperties['justifyItems'];
  alignItems?: CSSProperties['alignItems'];
  justifyContent?: CSSProperties['justifyContent'];
```

I would move these state props to the layout object itself, as they only make sense on the layout object (not the items)

Reply:
and the reason for adding templateRows, templateColumn naming , was just to make it a bit clear the connection to the css prop names